app/models/katello/concerns/setting_extensions.rb

```diff
@@ -17,7 +17,12 @@ def recalculate_errata_status
 
       def update_global_proxies
         if saved_change_to_attribute?(:value) && name == 'content_default_http_proxy'
-          repos = RootRepository.with_global_proxy.collect(&:library_instance).uniq
+          repos = []
+          RootRepository.with_global_proxy.each do |root_repo|
```

Comment:
Is this AR query equivalent to this logic you implemented here?

`Katello::Repository.joins(:root).merge(Katello::RootRepository.with_global_proxy).where.not(remote_href: nil).where(library_instance_id: nil)`

```
SELECT "katello_repositories".* FROM "katello_repositories" INNER JOIN "katello_root_repositories" ON 
"katello_root_repositories"."id" = "katello_repositories"."root_id" WHERE 
"katello_root_repositories"."http_proxy_policy" = 'global_default_http_proxy' AND 
"katello_repositories"."remote_href" IS NOT NULL AND "katello_repositories"."library_instance_id" IS NULL
```

Reply:
@jturel looks like it accomplishes the same thing, yeah

Reply:
Think I should substitute it in for this and the other similar spot?

Reply:
I realize that in the other case, `repositories_with_proxy` returns an array of repos. That one might need to be a bit different.